### ISSUE TITLE
Error with Promise Cancellation

### ISSUE THREAD

**Issue by p-j** [OptimalBits/bull#1129]
## Description
```Error: cannot enable cancellation after promises are in use```

This error has been introduced by this change: https://github.com/OptimalBits/bull/commit/365150f350ca4bdc5732898545f0a59d29c603b2#diff-b68fbb0f945cdf9cddec65ddf0fbe456

## Bull version
- 3.5.1 :heavy_multiplication_x: 
- 3.5.0 :heavy_multiplication_x: 
- 3.4.8 :heavy_check_mark: 

Previous occurrence:
https://github.com/OptimalBits/bull/issues/547

**Comment by alolis** [comment]
May i ask how are you reproducing this error exactly? Are you instantiating a `Promise` before importing bull maybe?

**Comment by manast** [comment]
Can please you provide an example so that we can reproduce the issue?

**Comment by p-j** [comment]
This is the same as https://github.com/OptimalBits/bull/issues/1130

app.js
```
const Promise = require('bluebird')
// do something with a Bluebird Promise
const bull = require('bull')
// → you get the error
```

**Comment by manast** [comment]
Fixed in version 3.5.2
